keps/sig-network/5343-kube-proxy-backends/README.md

```diff
@@ -0,0 +1,824 @@
+of the targeted release**.
+
+For enhancements that make changes to code or processes/procedures in core
+Kubernetes—i.e., [kubernetes/kubernetes], we require the following Release
+Signoff checklist to be completed.
+
+Check these off as they are completed for the Release Team to track. These
+checklist items _must_ be updated for the enhancement to be released.
+-->
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+With [KEP-3866] going GA, there are now 4 supported kube-proxy
+backends in-tree (`iptables`, `ipvs`, `nftables`, and `winkernel`).
+The nftables KEP noted that, as future work, we should figure if/how
+we can deprecate either or both of the older Linux backends. There has
+also been sporadic discussion about deprecating the `winkernel`
+backend or moving it out of tree, since it is not clear that it
+benefits from being in-tree.
+
+Meanwhile, an earlier KEP, [KEP-3786], had proposed moving most of the
+shared kube-proxy code into the kube-proxy staging repo
+(`k8s.io/kube-proxy`) so that it could be more easily shared by
+out-of-tree service proxy implementations. That KEP was never merged,
+though there was general agreement with the high-level idea.
```

Comment:
more than agreement is more about who the reasonable doubt about if is worth to do it ...`kpng` also thought there were a lot of people going to use a shared libraries that implemented service and talked about an existing demand for it ... but the anecdotal evidence is than more that a user story is a developer story and there was no such demand in the industry